flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/api/reconciler/ReconciliationMetadata.java

```diff
@@ -35,19 +34,15 @@ public class ReconciliationMetadata {
 
     private String apiVersion;
 
-    private ObjectMeta metadata;
```

Comment:
Is removing this a backward compatible change? Wouldn't Jackson fail if this field is there from a previous operator version that populated it? Can you please test / verify it?

Reply:
I'm removing the `metadata` node straight from the tree in [SpecUtils](https://github.com/apache/flink-kubernetes-operator/pull/834/files#diff-54298c2a18cbdf8528239150b0d9ea653c3a05c229f1cae62b3803593c68faaaR58), so the idea is that the generation field should not exist in the context of `ReconciliationMetadata`, but rather set independently as `observedGeneration` within `ReconciliationUtils` (sourced via `resource.getMetadata().getGeneration()`, which is different from the removed field).
I've also added a test for backwards compatibility, and verified tests pass on my forked branch.

Reply:
looks good!